x/axelarnet/keeper/msg_server.go

```diff
@@ -22,23 +25,67 @@ var _ types.MsgServiceServer = msgServer{}
 
 type msgServer struct {
 	Keeper
-	nexus   types.Nexus
-	bank    types.BankKeeper
-	account types.AccountKeeper
-	ibcK    IBCKeeper
+	nexus    types.Nexus
+	bank     types.BankKeeper
+	account  types.AccountKeeper
+	ibcK     IBCKeeper
+	multisig types.MultisigKeeper
 }
 
 // NewMsgServerImpl returns an implementation of the axelarnet MsgServiceServer interface for the provided Keeper.
-func NewMsgServerImpl(k Keeper, n types.Nexus, b types.BankKeeper, a types.AccountKeeper, ibcK IBCKeeper) types.MsgServiceServer {
+func NewMsgServerImpl(k Keeper, n types.Nexus, b types.BankKeeper, a types.AccountKeeper, ibcK IBCKeeper, ms types.MultisigKeeper) types.MsgServiceServer {
 	return msgServer{
-		Keeper:  k,
-		nexus:   n,
-		bank:    b,
-		account: a,
-		ibcK:    ibcK,
+		Keeper:   k,
+		nexus:    n,
+		bank:     b,
+		account:  a,
+		ibcK:     ibcK,
+		multisig: ms,
 	}
 }
 
+func (s msgServer) CallContract(c context.Context, req *types.CallContractRequest) (*types.CallContractResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	chain, ok := s.nexus.GetChain(ctx, req.Chain)
+	if !ok {
+		return nil, fmt.Errorf("%s is not a registered chain", req.Chain)
+	}
+
+	if !s.nexus.IsChainActivated(ctx, chain) {
+		return nil, fmt.Errorf("chain %s is not activated yet", chain.Name)
+	}
+
+	_, ok = s.multisig.GetCurrentKeyID(ctx, chain.Name)
```

Comment:
I would prefer do not check multisig key, the dest chain may not have multisig mechanism

Reply:
I see. If the dest chain does use multisig, and we don't check here, we need to check in the end blocker. And if the check fails in the end blocker, the message will never get sent to the destination chain, even though the transaction itself succeeded. Will the client be able to see this failure? Or is there a way to check if a chain does use multisig?

Reply:
I moved this check to the evm end blocker